### ISSUE TITLE
Error when visiting /admin/taxonomy

### ISSUE THREAD

**Issue by Trexology** [DevFactoryCH/taxonomy#8]
FatalErrorException in TaxonomyController.php line 17:
Class 'Devfactory\Taxonomy\Controllers\Controller' not found

Using Laravel 5.1
Taxonomy Branch 3.0

Please assist. thanks

**Comment by markcameron** [comment]
Hi, make sure you're using `"devfactory/taxonomy": "3.0.*"` and not `3.0.x-dev`. There's some work being done on the dev version that will break things.

**Comment by Trexology** [comment]
Yes i am using "devfactory/taxonomy": "3.0.*"

The installed version is 3.0.1

**Comment by offroff** [comment]
I installed this a minute ago and got almost the same error:
FatalErrorException in TaxonomyController.php line 17:
Class 'BaseController' not found

**Comment by offroff** [comment]
I tried to add 'BaseController' => Illuminate\Routing\Controller::class as an alias but then I got an error at line 23:
Cannot call constructor

Illuminate\Routing\Controller is an abstract class so that can't be constructed of course. I don't know which BaseController we need.

**Comment by Trexology** [comment]
Seem like there are some work going on.

The latest commit in 3.0 branch mention something about admin UI

**Comment by markcameron** [comment]
Hi all,

sorry for the delay in replying. So you guys have been sneaky and been trying to access the Admin interface that I made that is quite customized to our bootstrap admin theme (AdminLTE). I hadn't bothered getting it to work with Laravel 5 controller/views when I updated the package.

I spent a short time today fixing up what my colleague started the other day. Still needs work though (translations, testing). But if you use the `3.0.x@dev` branch, it should now load those pages, with interesting theming depending on your css.

To get it to work you'll need to add the following to your `routes.php` file:
```
Route::controller('taxonomy', '\Devfactory\Taxonomy\Controllers\TaxonomyController');
Route::controller('terms', '\Devfactory\Taxonomy\Controllers\TermsController');
```
and also edit the `config/taxonomy.config.php` file to use the correct `layout.extends` value that you put in your `@extends` for your blade template files.

The `route_prefix` value is no longer used I believe since you placed the routes above in your `routes.php` file, As opposed to setting the routes in the package route file like I did for Laravel 4. This allows you to apply middleware to the route anyway you wish rathen then me needing to bake in that functionality.

Let me know if you have any other problems.

**Comment by Trexology** [comment]
Hi @markcameron 

Thanks, I am able to load the UI now.

Spotted small errors when trying to load the Terms UI.

Inside terms/index.blade line:188 should be:
{!! Html::script('bla bla bla') !!}
instead of 
{!! HTML::script('bla bla bla') !!}

The next problem is:
vendor/taxonomy/js/jquery.nestable.js
does not seems to load at all, not sure if it is due to the path or something else that is preventing the loading.

**Comment by markcameron** [comment]
Sorry, forgot to mention that you need to publish the assets

`php artisan vendor:publish --tag=public`

that will but the nestable js file in the public folder

**Comment by sirsquall** [comment]
Hi,

I have created a new stable release 3.0.2, with the admin part, I have made few test with our admin theme everything look good, but could you try with yours and let me know if we need to do some update for you.

**Comment by Trexology** [comment]
Hi @sirsquall,

Just tried the new stable release, there are still abit of problem with the nestable javascript in Terms index.blade

line:177 to generate CSRF token in L5, somehow we have to use this 
'X-CSRF-Token': '{!! csrf_token() !!}',

line:189 should be: 
{!! Html::script('vendor/taxonomy/js/jquery.nestable.js') !!}

Thank you everyone for attending to my request! :D

P.S:
For those who wanna load the nestable js, I believe you have to add @yield('js') 
in your admin layout template

**Comment by sirsquall** [comment]
Hi,

Thanks I will update the way to get the csrf_token.

But I don't understand what is the change you need to fhe `HTML::script` ?

**Comment by Trexology** [comment]
@sirsquall 
By default, in laravel 5, Html alias are not embed anymore.

Actually, I'm not sure which whether 'HTML' or 'Html' is the more commonly used naming convention.

Assuming in L4 is named as 'HTML' then i think we should just stick to the old naming convention.

**Comment by sirsquall** [comment]
@Trexology Okey yes I will keep the old naming convention, I think is better.

I have released the 3.0.3 who fix the crsf.

Thanks
